keyboards/matrix/noah/keymaps/blockader/keymap.c

```diff
@@ -0,0 +1,657 @@
+                    }
+                    return false;
+                case KC_4:
+                    if (record->event.pressed) {
+                        layer_control_data.multiplier = layer_control_data.multiplier * 10 + 4;
+                        if (layer_control_data.multiplier < 0) layer_control_data.multiplier = 0;
+                    }
+                    return false;
+                case KC_5:
+                    if (record->event.pressed) {
+                        layer_control_data.multiplier = layer_control_data.multiplier * 10 + 5;
+                        if (layer_control_data.multiplier < 0) layer_control_data.multiplier = 0;
+                    }
+                    return false;
+                case KC_6:
+                    if (record->event.pressed) {
+                        layer_control_data.multiplier = layer_control_data.multiplier * 10 + 6;
+                        if (layer_control_data.multiplier < 0) layer_control_data.multiplier = 0;
+                    }
+                    return false;
+                case KC_7:
+                    if (record->event.pressed) {
+                        layer_control_data.multiplier = layer_control_data.multiplier * 10 + 7;
+                        if (layer_control_data.multiplier < 0) layer_control_data.multiplier = 0;
+                    }
+                    return false;
+                case KC_8:
+                    if (record->event.pressed) {
+                        layer_control_data.multiplier = layer_control_data.multiplier * 10 + 8;
+                        if (layer_control_data.multiplier < 0) layer_control_data.multiplier = 0;
+                    }
+                    return false;
+                case KC_9:
+                    if (record->event.pressed) {
+                        layer_control_data.multiplier = layer_control_data.multiplier * 10 + 9;
+                        if (layer_control_data.multiplier < 0) layer_control_data.multiplier = 0;
+                    }
+                    return false;
+                case KEY_INSERT_LINE_START:
+                    if (record->event.pressed) common_layer_data.back = true;
+                    return true;
+                case KEY_INSERT_HERE:
+                    if (record->event.pressed) common_layer_data.back = true;
+                    return true;
+                case KEY_INSERT_LINE_END:
+                    if (record->event.pressed) common_layer_data.back = true;
+                    return true;
+                case KEY_CREATE_PREVIOUS_LINE:
+                    if (record->event.pressed) common_layer_data.back = true;
+                    return true;
+                case KEY_CUT_WORD:
+                    if (record->event.pressed) {
+                        if (layer_control_data.operator== - 1) {
+                            SEND_STRING(SS_LALT(SS_TAP(X_LEFT) SS_TAP(X_RIGHT)));
```

Comment:
A lot of these `SEND_STRING`s could be converted to simple register/unregister/tap_code sequences, this will probably reduce the firmware size by a lot.

Reply:
Thanks, I will try this.

Reply:
Did you leave the remaining on purpose or just forgot them?

Reply:
Actually, I don't see why replacing SEND_STRING(SS_LALT(SS_TAP(X_LEFT) SS_TAP(X_RIGHT))) will reduce the firmware size. Previously there is one function call, now there are four function calls.

Reply:
If you can get rid of all the `SEND_STRING()` calls, theoretically the three LUTs and the code for the function itself will not get compiled in:
https://github.com/qmk/qmk_firmware/blob/master/quantum/quantum.c#L342-L541

Reply:
Compare the code for `register`/`unregister`/`tap_code16()`:
https://github.com/qmk/qmk_firmware/blob/master/quantum/quantum.c#L62-L111

Reply:
Okay! That's great!